src/main.rs

```diff
@@ -81,8 +81,17 @@ fn main() -> Result<(), Box<dyn Error>> {
         );
     let matches = match opts.parse(std::env::args()) {
         Ok(it) => it,
+        Err(getopts::Fail::UnrecognizedOption(string)) => {
+            println!(
+                "{}",
+                opts.usage(&format!(
+                    "Unknown option '{string}'! Please rerun with one of these:"
+                ))
+            );
+            // Quit the application with `EX_USAGE` from `sysexits.h`.
+            std::process::exit(64i32);
```

Comment:
Just a little note, in the last rust update (version `1.61`) the `Termination` trait landed: [doc](https://doc.rust-lang.org/std/process/trait.Termination.html), it allows you to create your own return type for the main function. This might be interesting for something like this. Maybe there is a crate on crates.io, that implements the `sysexits.h` header. Or we could create our own enum, with the different main return states.

But this is just thinking out loud, no change necessary, if you are not interested in this.

Reply:
I think `std::process::exit(64)` is more clear, since exit takes i32 only.

Reply:
There is indeed a crate therefore:  https://github.com/rust-lang/crates.io-index/blob/master/sy/se/sysexits.

And yes, the custom return type definitely sounds interesting.  How about a classic `i32`, just as in C?

Reply:
> I think `std::process::exit(64)` is more clear, since exit takes i32 only.

:wrench: :heavy_check_mark:

Reply:
> How about a classic `i32`, just as in C?

That would actually not work.. The `main` function can only return a `Result` of some sorts, or the unit type ("void" in c). And you can not implement that `Termination` trait for `i32` since both the trait, and the data type are defined in another module.

Reply:
Okay, that makes sense.  I would like to suggest to leave the current return type as it is and to continue the search of a different return type if required in the future.